spec/lib/mobile_application_platform/sign_up/service_spec.rb

```diff
@@ -0,0 +1,258 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'mobile_application_platform/sign_up/service'
+
+describe MobileApplicationPlatform::SignUp::Service do
+  let(:icn) { '10101V964144' }
+  let(:log_prefix) { '[MobileApplicationPlatform][SignUp][Service]' }
+
+  describe '#status' do
+    subject { described_class.new.status(icn:) }
```

Comment:
Using named subjects when defining an explicit subject is generally preferred because it adds context.

https://blog.davidchelimsky.net/blog/2012/05/13/spec-smell-explicit-use-of-subject/
https://github.com/rubocop/rubocop-rspec/issues/53#issuecomment-557729376

One improvement might be defining the service class as the subject in the top-level `describe` and then calling the methods you are testing in the nested `describe` blocks directly in the expectation. You can also have named subjects nested in a child block and referencing the named subject in the parent block, which reflects that the overall spec is testing the `MobileApplicationPlatform::SignUp::Service` spec and the nested blocks are testing the individual method calls.

```ruby
describe MobileApplicationPlatform::SignUp::Service do
  subject(:service) { described_class.new }
  ...

  describe '#status' do
    subject(:status_call) { service.status(:icn) }

    context 'when an issue occurs with the client request' do
      ...

      it 'raises a client error with expected message' do
        expect { status_call }.to raise_error(expected_error, expected_error_message)
      end
      ...
```

Reply:
That's interesting about named subjects. My general approach is to make the actual test as generic as possible, and I think the clarity comes from the context descriptions. 

But I could be convinced otherwise. Could be a good thing to bring up backend-wide just as a code quality conversation (we basically don't use named subjects in the vets-api codebase... but maybe we should change that?)